src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -220,9 +240,34 @@ private void RemovePlaceholderScope(BoundValuePlaceholderBase placeholder)
 
         private uint GetPlaceholderScope(BoundValuePlaceholderBase placeholder)
         {
-            Debug.Assert(_placeholderScopes is { });
-            return _placeholderScopes[placeholder];
+            Debug.Assert(_placeholderScopes?.ContainsKey(placeholder) == true);
+
+            return _placeholderScopes?.TryGetValue(placeholder, out var scope) == true
+                ? scope
+                : CallingMethodScope;
+        }
+
+#if DEBUG
+        private bool ContainsPlaceholderScope(BoundValuePlaceholderBase placeholder)
+        {
+            if (_placeholderScopes?.ContainsKey(placeholder) == true)
+            {
+                return true;
+            }
+
+            switch (placeholder)
```

Comment:
If I understand properly, we have a few grandfathered exceptions. Consider putting a comment here to that effect (we should strive not to add new ones)